ext/gd/gd.c

```diff
@@ -3633,6 +3633,11 @@ PHP_FUNCTION(imagescale)
 
 	im = php_gd_libgdimageptr_from_zval_p(IM);
 
+	if (tmp_h < 0 && tmp_w < 0) {
+		zend_value_error("$width and $height cannot be both negative");
```

Comment:
We typically construct an argument_value_error-esque wording when doing these kinds of things.
Something like: `Argument #2 ($width) and argument #3 ($height) cannot both be negative`.

Reply:
Ok will change to return false here instead and will do a PR just for master, wdyt ?

Reply:
Sounds good

Reply:
I think the patch as it's now doesn't change anything (well, we bail out a bit earlier), since previously the function also returned `false` if both width and height have been negative. I suggest to apply the previous patch to PHP-8.4+ (the ValueError was already there previously, only the error message changes). Only applying that to master is fine for me, too.

Reply:
Ah right indeed. I agree.